internal/integration/integration_test.go

```diff
@@ -321,8 +321,8 @@ func (suite *SpokesReceivePackTestSuite) TestSpokesReceivePackPushFromShallowClo
 		"git", "push", "--receive-pack=spokes-receive-pack-wrapper", "origin", "HEAD:test").CombinedOutput()
 
 	suite.T().Logf("ERROR:%s", out)
-	require.NoError(suite.T(), err)
-
+	require.Error(suite.T(), err)
+	require.Contains(suite.T(), string(out), "missing necessary objects")
```

Comment:
@migue Is it wrong for this test to fail?

Reply:
I don't see why this test should fail; it's just performing a push from a shallow clone:

```shell
git clone --depth=1 <remote-repo>
git push remote HEAD:test
```

I don't think this should fail?

Reply:
Hm, yeah, it should be OK.


```
$ git clone --depth=1 https://github.com/spraints/silver-eureka blah
Cloning into 'blah'...
remote: Enumerating objects: 5, done.
remote: Counting objects: 100% (5/5), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 5 (delta 1), reused 3 (delta 1), pack-reused 0
Receiving objects: 100% (5/5), 7.18 KiB | 7.18 MiB/s, done.
Resolving deltas: 100% (1/1), done.

$ cd blah

$ git push origin HEAD:refs/heads/anything
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
remote:
remote: Create a pull request for 'anything' on GitHub by visiting:
remote:      https://github.com/spraints/silver-eureka/pull/new/anything
remote:
To github.com:spraints/silver-eureka
 * [new branch]      HEAD -> anything
```

The test must not be set up quite right, I'll take a look.

Reply:
I figured out what the problem is... `spokes-receive-pack` assumes it's running in a bare repository, and this test is pushing to a non-bare repository. 452492ef4797d991892dbb063ea20e9f2fc8a893 changes the test so that it pushes to a bare repository (and adds a note that the repository is expected to be bare).